### PR TITLE
Release 2.3.3

### PR DESCRIPTION
**Pull request by lstein**
(note that this is actually release candidate 7, but I made the mistake of including an old rc number in the branch and can't easily change it)

## Updating Root directory

- Introduced new mechanism for updating the root directory when necessary. Currently only used to update the invoke.sh script using new dialog colors.
- Fixed ROCm torch module version number

## Loading legacy 2.0/2.1 models
- Due to not converting the torch.dtype precision correctly, the `load_pipeline_from_original_stable_diffusion_ckpt()` was returning models of dtype float32 regardless of the precision setting. This caused a precision mismatch crash.
- Problem now fixed (also see #3057 for the same fix to `main`)

## Support for a fourth textual inversion embedding file format
- This variant, exemplified by "easynegative.safetensors" has a single 'embparam' key containing a Tensor.
- Also refactored code to make it easier to read.
- Handle both pickle and safetensor formats.

## Persistent model selection
- To be consistent with WebUI parameter behavior, the currently selected model is saved on exit and restored on restart for both WebUI and CLI

## Bug fixes
- Name of VAE cache directory was "hug", not "hub". This is fixed.

## VAE fixes
- Allow custom VAEs to be assigned to a legacy model by placing a like-named vae file adjacent to the checkpoint file.
- The custom VAE will be picked up and incorporated into the diffusers model if the user chooses to convert/optimize.

## Custom config file loading
- Some of the civitai models instruct users to place a custom .yaml file adjacent to the checkpoint file. This generally wasn't working because some of the .yaml files use FrozenCLIPEmbedder rather than WeightedFrozenCLIPEmbedder, and our FrozenCLIPEmbedder class doesn't handle the `personalization_config` section used by the the textual inversion manager. Other .yaml files don't have the `personalization_config` section at all. Both these issues are fixed.#1685 

## Consistent pytorch version
- There was an inconsistency between the pytorch version requirement in `pyproject.toml` and the requirement in the installer (which does a little jiggery-pokery to load torch with the right CUDA/ROCm version prior to the main pip install. This was causing torch to be installed, then uninstalled, and reinstalled with a different version number. This is now fixed.

### CODE REVIEW

File: installer/lib/installer.py [hunk 1]
```diff
@@ -379,6 +383,9 @@ def install_user_scripts(self):
             shutil.copy(src, dest)
             os.chmod(dest, 0o0755)
 
+        if OS == "Linux":
+            shutil.copy(Path(__file__).parent / '..' / "templates" / "dialogrc", self.runtime / '.dialogrc')
```
Comment (ebr) [inline installer/lib/installer.py:387]:
```suggestion
            shutil.copy(Path(__file__).parents[1] / "templates" / "dialogrc", self.runtime / '.dialogrc')
```

**Comment by blessedcoolant** [comment]
Gave it a quick test. Seems good. Could only test the Windows stuff. LGTM.

**Review by blessedcoolant — APPROVED** [review]

File: ldm/modules/encoders/modules.py [hunk 2]
```diff
@@ -463,6 +463,10 @@ def forward(self, text, **kwargs):
     def encode(self, text, **kwargs):
         return self(text, **kwargs)
 
+    def set_textual_inversion_manager(self, manager): #TextualInversionManager):
+        # TODO all of the weighting and expanding stuff needs be moved out of this class
```
Comment (damian0815) [inline ldm/modules/encoders/modules.py:467]: i think this comment may be redundant? if i wrote it it's referring to stuff that Compel now handles

**Review by damian0815 — APPROVED** [review]
looks good to me

File: ldm/invoke/config/invokeai_update.py [hunk 3]
```diff
@@ -62,17 +65,26 @@ def main():
...
     elif choice=='2':
-        tag = 'main'
+        release = 'main'
     elif choice=='3':
-        tag = Prompt.ask('Enter an InvokeAI tag or branch name')
+        tag = Prompt.ask('Enter an InvokeAI tag name')
+    elif choice=='4':
+        branch = Prompt.ask('Enter an InvokeAI branch name')
 
     print(f':crossed_fingers: Upgrading to [yellow]{tag}[/yellow]')
```
Comment (ebr) [inline ldm/invoke/config/invokeai_update.py:81]: image


```suggestion
    print(f':crossed_fingers: Upgrading to [yellow]{tag if tag else release}[/yellow]')
```

**Review by ebr — APPROVED** [review]
tested install / update process - all working well aside from a minor updater fix (see suggestion) - approving anyway to unblock the merge

File: ldm/modules/encoders/modules.py (hunk 2 above)
Comment (lstein) [reply ldm/modules/encoders/modules.py:467]: Removed the comment.

**Pull request merged**